### PR TITLE
python3*: revert asyncio changes done in 3.10.9 and 3.11.1

### PR DESCRIPTION
**Pull request by vcunat**
They brought significant regressions.  Upstream is now discussing what to do, but we still want the security fixes from 3.10.9.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
`python3Packages.websockets` confirmed to build again.  Other builds in progress.  (x86_64-linux)

**Review by SuperSandro2000 — APPROVED** [review]
IMO a better choice than reverting the update and cherry-picking the fixes. Lets hope upstream gets this resolved soonish.

File: pkgs/development/interpreters/python/cpython/default.nix [hunk 1]
```diff
@@ -214,7 +214,10 @@ in with passthru; stdenv.mkDerivation {
     substituteInPlace setup.py --replace /Library/Frameworks /no-such-path
   '';
 
-  patches = [
+  patches = optionals (version == "3.10.9") [
```
Comment (mweinelt) [inline pkgs/development/interpreters/python/cpython/default.nix:217]:
```suggestion
  patches = optionals (version == "3.10.9" || version == "3.11.1") [
```

Python 3.11 is also affected.

**Comment by mweinelt** [comment]
Had been discussing this approach with some people. Thanks for beating me to the punch.

File: pkgs/development/interpreters/python/cpython/default.nix (hunk 1 above)
Comment (vcunat) [reply pkgs/development/interpreters/python/cpython/default.nix:217]: Ah, I thought it wasn't updated as I didn't see it in the same MR.  But this patch won't apply there, unfortunately.

Comment (vcunat) [reply pkgs/development/interpreters/python/cpython/default.nix:217]: 3.11 at least isn't so common yet, so it should be way smaller rebuild and thus could be fixed later.  I expect we might simply `fetchpatch` https://github.com/python/cpython/commit/3fae04b10e2655a20a3aadb5e0d63e87206d0c67 separately for 3.11.

Comment (mweinelt) [reply pkgs/development/interpreters/python/cpython/default.nix:217]: Fine, we can leave it as is and unblock just python310 right now.

**Comment by mweinelt** [comment]
Nit: The patch should go into the cpython/3.10 directory.

**Comment by vcunat** [comment]
OK, moved and added patch for 3.11 as well.  Though I'm not sure there, maybe python 3.11 packages are broken overall so far.

**Comment by vcunat** [comment]
Yes, I see the same build failures with 3.11.0, so I guess I'll just assume that the patch works.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**